src/lib.rs

```diff
@@ -156,7 +156,7 @@ async fn handle_ohttp_relay(
 #[instrument]
 fn into_forward_req(
     mut req: Request<Incoming>,
-    gateway_origin: &Uri,
+    gateway_origin: &GatewayUri,
```

Comment:
Why is this the most appropriate place to take the type as an argument rather than at `ohttp_relay` other than "that's where it was done before and I didn't want to make excess changes"